Sources/ApodiniDeploy/DeploymentOptions/Condition.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  File.swift
+//  
+//
+//  Created by Lukas Kollmer on 2021-02-06.
+//
+
+import Foundation
```

Comment:
Yes exactly. The idea is as follows:
- ApodiniDeploy: the "user-facing" target. a web service would import this, as well as all deployment providers it wants to use
- ApodiniDeployBuildSupport: everything shared between ApodiniDeploy and ApodiniDeployRuntimeSupport. This is the target one would import to implement a deployment provider. (eg you'll see that neither of the deployment providers imports the ApodiniDeploy target, they import only ApodiniDeployBuildSupport and ApodiniDeployRuntimeSupport

I think moving the conditions into ApodiniDeployBuildSupport should be possible, but that would require us to make either `AnyOption.resolve` or `ConditionalOption` public (because the interface exporter needs to access these to resolve the conditions), which in turn would also make them public to the web service importing ApodiniDeploy. I wasn't sure if that would be a good idea, since the resolve function and the ConditionalOption types are implementation details which are not relevant to (and should not be used by) the user.

Reply:
If I understood this correctly the functionality of `ConditionalOption` is not really needed in `ApodiniDeployBuildSupport` because you can define an option without the need to implement anything on top to make it conditional. So every option that is defined can be made conditional and that functionality is only needed in ApodiniDeploy. If I understood this correctly I am fine with keeping it in the ApodiniDeploy target 👍

Reply:
Yeah pretty much. There's an extension `when(_ condition: AnyHandlerCondition)` on `AnyOption`, which allows you to turn any option into a conditional option (eg: `.memory(.mb(12)).when(...)`).
I initially defined the API in a way that you'd put the condition directly into the option (eg: `.memory(.mb(12), when: ...)`), but that'd require the person defining a new option to also add the necessary overloads to support conditions, which just results in a lot of unnecessary boilerplate code